### PR TITLE
Gems implementation

### PR DESCRIPTION
**Pull request by geraldma1998**
- Added and implemented devise authentication for User model.
- Added rspec gem.
- Added factory_bot gem.
- Added faker gem.
- Added shoulda-matchers gem.
- Added rubocop gem.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -53,3 +56,9 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+gem 'devise'
+gem 'faker', :git => 'https://github.com/stympy/faker.git', :branch => 'master'
+gem 'rubocop', require: false
```
Comment (maqr1993) [inline Gemfile]: You are missing the rubocop files. Create `.pernix-rubocop.yml` and `.rubocop.yml` files I'll share with you the content of each file.

File: Gemfile [hunk 2]
```diff
@@ -53,3 +56,9 @@ end
...
+gem 'bootstrap', '~> 4.3.1'
+gem 'jquery-rails'
```
[the remark this answers is not in the document]
Comment (geraldma1998) [reply Gemfile]: Done

File: Gemfile [hunk 3]
```diff
@@ -53,3 +56,9 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
```
[the remark this answers is not in the document]
Comment (geraldma1998) [reply Gemfile]: Done

File: app/controllers/concerns/devise_whitelist.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module DeviseWhitelist
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :configure_permited_parameters, if: :devise_controller?
+  end
+
+  def configure_permited_parameters
```
[the remark this answers is not in the document]
Comment (geraldma1998) [reply app/controllers/concerns/devise_whitelist.rb]: Done

File: app/helpers/home_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module HomeHelper
```
[the remark this answers is not in the document]
Comment (geraldma1998) [reply app/helpers/home_helper.rb]: Done

File: Gemfile (hunk 1 above)
Comment (geraldma1998) [reply Gemfile]: Done

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+	include DeviseWhitelist
```
[the remark this answers is not in the document]
Comment (geraldma1998) [reply app/controllers/application_controller.rb]: Done

**Pull request merged**